lib/quby/answers/services/updates_answers.rb

```diff
@@ -49,6 +46,20 @@ def on_failure(&block)
 
         protected
 
+        def save_raw_params_on_error(new_attributes, &_block)
```

Comment:
Het punt is dat `Quby.answers.update!` in een `rescue` staat. Het is dus "save raw params **only** on error". Het enige stuk dat onvoorwaardelijk wordt gedaan is:
```ruby
answer.raw_params = new_attributes
backup_answer = answer.clone
```
Niets dat de raw_params *savet*.

Maar ik begrijp je punt ergens wel, de functie geeft answer.raw_params wel altijd een nieuwe waarde maar slaat hem niet altijd op. Misschien is de flow wat duidelijker met zoiets:
```ruby
        def update(new_attributes = {})
          ensuring_raw_params_are_saved(new_attributes) do
            if answer.errors.empty?
              ...
              # GEEN actie die het record opslaat
              succeed!
            else
              fail!
            end
          end
        end
        
        def ensuring_raw_params_are_saved(new_attributes)
           answer.raw_params = new_attributes
           backup_answer = answer.clone
  
           yield
        ensure
          backup_answer.raw_params['could_not_update_at'] = DateTime.now.to_i # to note this is was a failed update
          Quby.answers.update!(backup_answer)
        end
```
Weet alleen niet zeker of er dingen stuk gaan als `succeed!` wordt aangeroepen voordat het record gesaved is.

P.S. 🤣